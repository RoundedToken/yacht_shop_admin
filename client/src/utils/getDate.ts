export function getDate(isoDate: string): string {
    const date = new Date(isoDate);

    return date.toLocaleString('default', {
        day: '2-digit',
        month: 'short',
        year: 'numeric',
        hour: 'numeric',
        minute: 'numeric',
        hour12: false,
    });
}
