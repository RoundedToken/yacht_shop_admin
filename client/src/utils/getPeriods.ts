const toIso = (mils: number) => {
    const now = Date.now();

    return new Date(now - mils).toISOString();
};
const getThisWeek = () => {
    const today = new Date();

    const dayOfWeek = today.getDay() || 7;

    const monday = new Date(today.getTime() - (dayOfWeek - 1) * 24 * 60 * 60 * 1000);

    return monday.toISOString();
};

const getThisMonth = () => {
    const today = new Date();

    const year = today.getFullYear();
    const month = today.getMonth();

    const firstDay = new Date(year, month, 1, 23, 59);

    return firstDay.toISOString();
};

const getThisYear = () => {
    const today = new Date();

    const year = today.getFullYear();

    const firstDay = new Date(year, 0, 1, 23, 59);

    return firstDay.toISOString();
};

export const getPeriods = () => {
    const today = new Date().toISOString();
    const thisWeek = getThisWeek();
    const week = toIso(1000 * 60 * 60 * 24 * 7);
    const month = toIso(1000 * 60 * 60 * 24 * 30);
    const thisMonth = getThisMonth();
    const thisYear = getThisYear();
    const year = toIso(1000 * 60 * 60 * 24 * 365);

    return { week, month, year, today, thisWeek, thisMonth, thisYear };
};
