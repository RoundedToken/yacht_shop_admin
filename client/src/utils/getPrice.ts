export const getPrice = (price: number) => {
    const formatter = new Intl.NumberFormat('et-EE', {
        style: 'currency',
        currency: 'EUR',
    });
    return formatter.format(price);
};
