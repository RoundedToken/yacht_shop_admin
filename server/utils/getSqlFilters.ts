import { IFilters } from '../types';

export const getSqlFilters = (filters: string, isProducts: boolean = false) => {
    const filtersObj = JSON.parse(filters) as IFilters;
    const productId = filtersObj.product;

    const filterStr =
        Object.entries(filtersObj)
            .filter((filter) => filter[1] !== null && filter[1] !== '')
            .map((filter) => {
                const key = filter[0];
                const value = filter[1];

                if (key === 'dateFrom') {
                    return `ord.DateEnter >= CAST('${value}T00:00:00' AS datetime)`;
                } else if (key === 'dateTo') {
                    return `ord.DateEnter <= CAST('${value}T23:59:59' AS datetime)`;
                } else if (key === 'status') {
                    return `ord.status IN (${value.join(',')})`;
                } else if (key === 'isPaid') {
                    return `ord.paid = ${value ? 1 : 0}`;
                } else if (key === 'product') {
                    if (isProducts) return '1=1';
                    else return `ordStr.tovar = ${value}`;
                } else if (key === 'client') {
                    return `ord.clientid = ${value}`;
                } else {
                    return `STUFF((SELECT DISTINCT ', ' + oplata.type  
                    FROM Oplata 
                    WHERE Oplata.orderid = ord.orderid  
                    FOR XML PATH('')), 1, 2, '') LIKE '%${value}%'`;
                }
            })
            .join(' AND ') || '1=1';

    return { filterStr, productId };
};
