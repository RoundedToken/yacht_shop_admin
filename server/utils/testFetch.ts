export class testFetch {
    private endpoint: string;

    constructor(endpoint: string) {
        this.endpoint = endpoint;
    }

    async getData(params: Record<string, string>) {
        const baseUrl = 'http://192.168.1.37:5000/api';
        const url = baseUrl + this.endpoint + '?' + new URLSearchParams(params);

        const response = await fetch(url);

        return await response.json();
    }
}
