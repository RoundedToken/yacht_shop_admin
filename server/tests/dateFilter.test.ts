import { describe, expect, test } from 'bun:test';
import { testFetch } from '../utils/testFetch';

describe('Date Filter', () => {
    const getSales = new testFetch('/get_sales');
    const getProducts = new testFetch('/get_products');
    const params = {
        pageIndex: '0',
        pageSize: '50',
        sortId: 'Id',
        sortType: 'DESC',
        filters:
            '{"dateFrom":"2022-08-01","dateTo":"2022-08-04","payType":null,"status":[1,-1],"isPaid":null,"product":null,"client":null}',
    };

    test('Get Sales from 2022-08-01 to 2022-08-04', async () => {
        const data = await getSales.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Products from 2022-08-01 to 2022-08-04', async () => {
        const data = await getProducts.getData(params);

        expect(data).toMatchSnapshot();
    });
});
