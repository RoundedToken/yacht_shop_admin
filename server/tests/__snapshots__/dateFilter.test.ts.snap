// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Filter Get Sales from 2022-08-01 to 2022-08-04 1`] = `
{
  "clients": [
    {
      "id": null,
      "name": null,
    },
    {
      "id": 245,
      "name": "Щербаков Антон Васильевич",
    },
    {
      "id": 627,
      "name": "Sailtech Oy (Финляндия)",
    },
    {
      "id": 676,
      "name": "Maritim-Finland",
    },
    {
      "id": 940,
      "name": "web",
    },
    {
      "id": 1108,
      "name": "TREM",
    },
    {
      "id": 1702,
      "name": "MarineTech (Germany)",
    },
    {
      "id": 2766,
      "name": "Optiparts",
    },
    {
      "id": 2773,
      "name": "Windex (Sweden)",
    },
    {
      "id": 9901,
      "name": "Anton Scherbakov",
    },
    {
      "id": 11182,
      "name": "_jamuuk",
    },
    {
      "id": 11183,
      "name": "PJK",
    },
    {
      "id": 11185,
      "name": "Regate (LV)",
    },
    {
      "id": 11186,
      "name": "TommyyLyy",
    },
    {
      "id": 11187,
      "name": "Pro Marine Trade",
    },
    {
      "id": 11188,
      "name": "Татьяна Каверина (Armastus)",
    },
    {
      "id": 11189,
      "name": "NORA LII SOPRADE SELTS",
    },
    {
      "id": 11190,
      "name": "Andres",
    },
    {
      "id": 11191,
      "name": "Nautix",
    },
    {
      "id": 11192,
      "name": "Torivallavalitsus",
    },
    {
      "id": 11193,
      "name": "Ninne",
    },
    {
      "id": 11194,
      "name": "Urmas Varris",
    },
    {
      "id": 11195,
      "name": "WindWalker",
    },
    {
      "id": 11196,
      "name": "Via Grupp OU (Командор)",
    },
    {
      "id": 11201,
      "name": "Tõnis",
    },
    {
      "id": 11202,
      "name": "Jaanus",
    },
    {
      "id": 11203,
      "name": "Meister Arle OU",
    },
    {
      "id": 11204,
      "name": "Prime Marine Service OU",
    },
    {
      "id": 11205,
      "name": "Viking",
    },
    {
      "id": 11206,
      "name": "Saku",
    },
    {
      "id": 11207,
      "name": "SineneLind",
    },
    {
      "id": 11208,
      "name": "Alex Tido",
    },
    {
      "id": 11209,
      "name": "AZZURRA",
    },
    {
      "id": 11210,
      "name": "квас на пиво",
    },
    {
      "id": 11211,
      "name": "MIA (Tallin)",
    },
    {
      "id": 11212,
      "name": "MINNI",
    },
    {
      "id": 11213,
      "name": "Jonпgert OU",
    },
    {
      "id": 11214,
      "name": "MIA Pärnu",
    },
    {
      "id": 11217,
      "name": "Tauno Teluk",
    },
    {
      "id": 11219,
      "name": "Капитан",
    },
    {
      "id": 11220,
      "name": "Johannes  Atlandi Võlg",
    },
    {
      "id": 11221,
      "name": "Nimi",
    },
    {
      "id": 11222,
      "name": "Viahooldus ОÜ",
    },
    {
      "id": 11223,
      "name": "Kaupmees",
    },
    {
      "id": 11224,
      "name": "Reino",
    },
    {
      "id": 11225,
      "name": "Rubit OU",
    },
    {
      "id": 11226,
      "name": "Aleksey  Rusetsky",
    },
    {
      "id": 11227,
      "name": "Corinthian Events OÜ",
    },
    {
      "id": 11228,
      "name": "Ronstan DK",
    },
    {
      "id": 11229,
      "name": "Narva-Joesuu PurjeSpoordiKool",
    },
    {
      "id": 11230,
      "name": "Saaremaa Meri Spordiselts",
    },
    {
      "id": 11231,
      "name": "Tallux Orto",
    },
    {
      "id": 11232,
      "name": "Uku Kuusk",
    },
    {
      "id": 11234,
      "name": "Bainbridge Inc (UK)",
    },
    {
      "id": 11235,
      "name": "BetaMarine (UK)",
    },
    {
      "id": 11236,
      "name": "Change Consulting OÜ",
    },
    {
      "id": 11237,
      "name": "Rait Maru",
    },
    {
      "id": 11239,
      "name": "Žaklar OU",
    },
    {
      "id": 11240,
      "name": "Comar Systems Limited",
    },
    {
      "id": 11241,
      "name": "Kaidor",
    },
    {
      "id": 11242,
      "name": "MADEN",
    },
    {
      "id": 11243,
      "name": "Ropebox Eesti OÜ",
    },
    {
      "id": 11244,
      "name": "Muri",
    },
    {
      "id": 11245,
      "name": "Carrestoration OU",
    },
    {
      "id": 11246,
      "name": "Siren Sailing OU",
    },
    {
      "id": 11247,
      "name": "Episood OU",
    },
    {
      "id": 11248,
      "name": "Akster Grupp OU",
    },
    {
      "id": 11249,
      "name": "Alter",
    },
    {
      "id": 11250,
      "name": "Ronstan AUS",
    },
    {
      "id": 11251,
      "name": "Innon Finance OU",
    },
    {
      "id": 11252,
      "name": "Lettore OÜ",
    },
    {
      "id": 11253,
      "name": "Navideep Services OÜ",
    },
    {
      "id": 11254,
      "name": "Athaman (Polska)",
    },
    {
      "id": 11255,
      "name": "Lõbusõit OÜ",
    },
    {
      "id": 11256,
      "name": "Aleksei Mikhalev",
    },
    {
      "id": 11257,
      "name": "Ropebox OÜ",
    },
    {
      "id": 11258,
      "name": "Kaubakeskus OÜ",
    },
    {
      "id": 11259,
      "name": "Rein Ottoson OÜ",
    },
    {
      "id": 11260,
      "name": "Robert",
    },
    {
      "id": 11261,
      "name": "Merema OÜ",
    },
    {
      "id": 11262,
      "name": "Alexander Rusetsky",
    },
    {
      "id": 11263,
      "name": "Natalja Rusetskaya",
    },
    {
      "id": 11264,
      "name": "Jahtklubi Dago",
    },
    {
      "id": 11265,
      "name": "Elev Luik",
    },
    {
      "id": 11266,
      "name": "_ЗХ2023",
    },
    {
      "id": 11267,
      "name": "Vsevolod pyramidaprint",
    },
    {
      "id": 11268,
      "name": "Emerald",
    },
    {
      "id": 11270,
      "name": "Strand Sailing Management",
    },
    {
      "id": 11271,
      "name": "Tuule",
    },
    {
      "id": 11272,
      "name": "Meelis Kojamess (Баба Яга)",
    },
    {
      "id": 11273,
      "name": "Andrea FE64",
    },
    {
      "id": 11274,
      "name": "Reilut OÜ",
    },
    {
      "id": 11275,
      "name": "Pärnu cruises",
    },
    {
      "id": 11280,
      "name": "Madis Parts",
    },
    {
      "id": 11281,
      "name": "Tera ehitus OU",
    },
    {
      "id": 11282,
      "name": "Korkvender OU",
    },
    {
      "id": 11291,
      "name": "STRAND PROJEKT OÜ",
    },
    {
      "id": 11292,
      "name": "RTR Tehnika",
    },
    {
      "id": 11293,
      "name": "Eprodukt  OÜ",
    },
    {
      "id": 11294,
      "name": "Madis Parts",
    },
    {
      "id": 11295,
      "name": "SeaKayaking",
    },
    {
      "id": 11296,
      "name": "8 Motor Oy",
    },
    {
      "id": 11297,
      "name": "SKELTMER OÜ",
    },
    {
      "id": 11298,
      "name": "Jäälind",
    },
    {
      "id": 11299,
      "name": "MattChem (France)",
    },
    {
      "id": 11300,
      "name": "Car Restoration OU",
    },
    {
      "id": 11301,
      "name": "Valgeranna Seikluspark OÜ",
    },
    {
      "id": 11302,
      "name": "Kaif OÜ",
    },
    {
      "id": 11303,
      "name": "Julechka",
    },
    {
      "id": 11304,
      "name": "Longboat",
    },
    {
      "id": 11305,
      "name": "SAILING MASSIVE  OÜ",
    },
    {
      "id": 11306,
      "name": "AB Technology",
    },
    {
      "id": 11307,
      "name": "Norman Saarso",
    },
    {
      "id": 11308,
      "name": "Tuisu OÜ",
    },
    {
      "id": 11309,
      "name": "Fairaway",
    },
    {
      "id": 11311,
      "name": "Alibi OÜ",
    },
    {
      "id": 11312,
      "name": "Dirham motorpaad",
    },
    {
      "id": 11313,
      "name": "Niined group",
    },
    {
      "id": 11314,
      "name": "Magna Fide",
    },
    {
      "id": 11315,
      "name": "Корячкин",
    },
    {
      "id": 11317,
      "name": "Konstrado",
    },
    {
      "id": 11318,
      "name": "Vektom OÜ",
    },
    {
      "id": 11319,
      "name": "Kalev Jahtklubi",
    },
    {
      "id": 11320,
      "name": "Marta Vintage OÜ",
    },
  ],
  "orders": [
    {
      "allSum": 411.2021,
      "clientName": "Anton Scherbakov",
      "date": "2022-08-04T10:41:00.000Z",
      "id": 205098,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 12,
      "productsCount": -26,
      "productsSum": 411.2021,
      "status": "продажа",
    },
    {
      "allSum": 2.1,
      "clientName": "_jamuuk",
      "date": "2022-08-04T21:43:00.000Z",
      "id": 205060,
      "isPaid": true,
      "paidSum": 2.1,
      "payType": "Visa",
      "positionCount": 2,
      "productsCount": -2,
      "productsSum": 2.1,
      "status": "продажа",
    },
    {
      "allSum": 15,
      "clientName": "_jamuuk",
      "date": "2022-08-04T19:13:00.000Z",
      "id": 205059,
      "isPaid": true,
      "paidSum": 15,
      "payType": "Visa",
      "positionCount": 3,
      "productsCount": -6,
      "productsSum": 15,
      "status": "продажа",
    },
    {
      "allSum": 13,
      "clientName": "_jamuuk",
      "date": "2022-08-04T19:10:00.000Z",
      "id": 205058,
      "isPaid": true,
      "paidSum": 13,
      "payType": "Visa",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 13,
      "status": "продажа",
    },
    {
      "allSum": 49,
      "clientName": "_jamuuk",
      "date": "2022-08-04T16:50:00.000Z",
      "id": 205057,
      "isPaid": true,
      "paidSum": 49,
      "payType": "Visa",
      "positionCount": 2,
      "productsCount": -2,
      "productsSum": 49,
      "status": "продажа",
    },
    {
      "allSum": 18.67,
      "clientName": "Lettore OÜ",
      "date": "2022-08-04T15:25:00.000Z",
      "id": 205056,
      "isPaid": true,
      "paidSum": 18.67,
      "payType": "Visa",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 18.67,
      "status": "продажа",
    },
    {
      "allSum": 43,
      "clientName": "_jamuuk",
      "date": "2022-08-04T14:43:00.000Z",
      "id": 205055,
      "isPaid": true,
      "paidSum": 43,
      "payType": "Visa",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 43,
      "status": "продажа",
    },
    {
      "allSum": 12.6,
      "clientName": "_jamuuk",
      "date": "2022-08-04T14:36:00.000Z",
      "id": 205054,
      "isPaid": true,
      "paidSum": 12.6,
      "payType": "Visa",
      "positionCount": 2,
      "productsCount": -7,
      "productsSum": 12.6,
      "status": "продажа",
    },
    {
      "allSum": 12,
      "clientName": "_jamuuk",
      "date": "2022-08-03T19:38:00.000Z",
      "id": 205052,
      "isPaid": true,
      "paidSum": 12,
      "payType": "Visa",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 12,
      "status": "продажа",
    },
    {
      "allSum": 62,
      "clientName": "_jamuuk",
      "date": "2022-08-03T21:53:00.000Z",
      "id": 205051,
      "isPaid": true,
      "paidSum": 62,
      "payType": "Visa",
      "positionCount": 5,
      "productsCount": -5,
      "productsSum": 62,
      "status": "продажа",
    },
    {
      "allSum": 5,
      "clientName": "WindWalker",
      "date": "2022-08-03T18:07:00.000Z",
      "id": 205050,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 1,
      "productsCount": -2,
      "productsSum": 5,
      "status": "продажа",
    },
    {
      "allSum": 32,
      "clientName": "_jamuuk",
      "date": "2022-08-03T16:57:00.000Z",
      "id": 205048,
      "isPaid": true,
      "paidSum": 32,
      "payType": "Visa",
      "positionCount": 1,
      "productsCount": -2,
      "productsSum": 32,
      "status": "продажа",
    },
    {
      "allSum": 99,
      "clientName": "Lõbusõit OÜ",
      "date": "2022-08-03T12:56:00.000Z",
      "id": 205047,
      "isPaid": true,
      "paidSum": 99,
      "payType": "Б/н",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 99,
      "status": "продажа",
    },
    {
      "allSum": 12.7,
      "clientName": "_jamuuk",
      "date": "2022-08-03T12:33:00.000Z",
      "id": 205046,
      "isPaid": true,
      "paidSum": 9.1,
      "payType": "Visa",
      "positionCount": 4,
      "productsCount": -4,
      "productsSum": 12.7,
      "status": "продажа",
    },
    {
      "allSum": 0.4,
      "clientName": "_jamuuk",
      "date": "2022-08-04T16:54:00.000Z",
      "id": 205045,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 1,
      "productsCount": -2,
      "productsSum": 0.4,
      "status": "продажа",
    },
    {
      "allSum": 78,
      "clientName": "NORA LII SOPRADE SELTS",
      "date": "2022-08-01T09:56:00.000Z",
      "id": 204919,
      "isPaid": true,
      "paidSum": 78,
      "payType": "Б/н",
      "positionCount": 1,
      "productsCount": -31.2,
      "productsSum": 78,
      "status": "продажа",
    },
    {
      "allSum": 155.8,
      "clientName": "_jamuuk",
      "date": "2022-08-03T18:10:00.000Z",
      "id": 204903,
      "isPaid": true,
      "paidSum": 155.8,
      "payType": "Visa, Нал",
      "positionCount": 11,
      "productsCount": -80,
      "productsSum": 155.8,
      "status": "продажа",
    },
  ],
  "pagesCount": 1,
  "rowsCount": 17,
  "totalAllSum": 1021.4721,
  "totalPaidSum": 601.27,
  "totalProductSum": 1021.4721,
}
`;

exports[`Date Filter Get Products from 2022-08-01 to 2022-08-04 1`] = `
{
  "pagesCount": 1,
  "products": [
    {
      "avgPrice": 2.5,
      "id": 58172,
      "maxPrice": 2.5,
      "minPrice": 2.5,
      "name": "Olu Saku Rock  0.568",
      "ordersCount": 1,
      "totalCount": -15,
      "totalSum": 37.5,
    },
    {
      "avgPrice": 2,
      "id": 58171,
      "maxPrice": 2,
      "minPrice": 2,
      "name": "Kokteil Somersby apple 0.33",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 2,
    },
    {
      "avgPrice": 2.5,
      "id": 58169,
      "maxPrice": 2.5,
      "minPrice": 2.5,
      "name": "Olu Urquell Pilsner hele  4.4%  0.5",
      "ordersCount": 2,
      "totalCount": -9,
      "totalSum": 22.5,
    },
    {
      "avgPrice": 2.5,
      "id": 58168,
      "maxPrice": 2.5,
      "minPrice": 2.5,
      "name": "Olu Meistrite Gildi 4.8%  0.5",
      "ordersCount": 2,
      "totalCount": -4,
      "totalSum": 10,
    },
    {
      "avgPrice": 1.4,
      "id": 58167,
      "maxPrice": 1.4,
      "minPrice": 1.4,
      "name": "oriGINinal 0%  0.33",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 2.8,
    },
    {
      "avgPrice": 2.5,
      "id": 58094,
      "maxPrice": 2.5,
      "minPrice": 2.5,
      "name": "Olu Heineken 0.5",
      "ordersCount": 3,
      "totalCount": -10,
      "totalSum": 25,
    },
    {
      "avgPrice": 2,
      "id": 58027,
      "maxPrice": 2,
      "minPrice": 2,
      "name": "Kokteil Koff Grapefruit 0.33",
      "ordersCount": 1,
      "totalCount": -5,
      "totalSum": 10,
    },
    {
      "avgPrice": 2,
      "id": 58025,
      "maxPrice": 2,
      "minPrice": 2,
      "name": "Olu Saku Kuld 0.33",
      "ordersCount": 1,
      "totalCount": -21,
      "totalSum": 42,
    },
    {
      "avgPrice": 2,
      "id": 58024,
      "maxPrice": 2,
      "minPrice": 2,
      "name": "Olu Saku Originaal 0.33",
      "ordersCount": 1,
      "totalCount": -3,
      "totalSum": 6,
    },
    {
      "avgPrice": 1,
      "id": 58021,
      "maxPrice": 1,
      "minPrice": 1,
      "name": "Vesi Vichi Still  vesi 0.5L PET 24",
      "ordersCount": 1,
      "totalCount": -15,
      "totalSum": 15,
    },
    {
      "avgPrice": 1,
      "id": 58020,
      "maxPrice": 1,
      "minPrice": 1,
      "name": "Vesi Vichi Mineraal vesi 0.5L PET 12",
      "ordersCount": 1,
      "totalCount": -3,
      "totalSum": 3,
    },
    {
      "avgPrice": 12,
      "id": 56078,
      "maxPrice": 12,
      "minPrice": 12,
      "name": "Резьбовая часть талрепа вилка-кач левая М12",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 12,
    },
    {
      "avgPrice": 12,
      "id": 56071,
      "maxPrice": 12,
      "minPrice": 12,
      "name": "Карабин самостопор 12mm A4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 12,
    },
    {
      "avgPrice": 30,
      "id": 56069,
      "maxPrice": 30,
      "minPrice": 30,
      "name": "Перчатки sailing SF gloves",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 30,
    },
    {
      "avgPrice": 1.5,
      "id": 56057,
      "maxPrice": 1.5,
      "minPrice": 1.5,
      "name": "Carbic 4mm красная",
      "ordersCount": 1,
      "totalCount": -4,
      "totalSum": 6,
    },
    {
      "avgPrice": 19,
      "id": 55208,
      "maxPrice": 19,
      "minPrice": 19,
      "name": "Кепка Sailing Ash One Size",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 19,
    },
    {
      "avgPrice": 3.1,
      "id": 52985,
      "maxPrice": 3.1,
      "minPrice": 3.1,
      "name": "Палец стопорной 12x30 А4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 3.1,
    },
    {
      "avgPrice": 28,
      "id": 49712,
      "maxPrice": 28,
      "minPrice": 28,
      "name": "Талреп закрытый вилка-обжим M12/6mm A4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 28,
    },
    {
      "avgPrice": 5.292,
      "id": 47829,
      "maxPrice": 5.292,
      "minPrice": 5.292,
      "name": "Дужка на средний стопор желтая",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 10.584,
    },
    {
      "avgPrice": 5.292,
      "id": 47827,
      "maxPrice": 5.292,
      "minPrice": 5.292,
      "name": "Дужка на средний стопор красная",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 10.584,
    },
    {
      "avgPrice": 2.5,
      "id": 44697,
      "maxPrice": 2.5,
      "minPrice": 2.5,
      "name": "Веревка шкотовая  SAN DIEGO 10mm красная",
      "ordersCount": 1,
      "totalCount": -31.2,
      "totalSum": 78,
    },
    {
      "avgPrice": 43,
      "id": 41731,
      "maxPrice": 43,
      "minPrice": 43,
      "name": "Ветроуказатель Windex 10"С",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 43,
    },
    {
      "avgPrice": 4.728,
      "id": 38227,
      "maxPrice": 4.728,
      "minPrice": 4.728,
      "name": "Дужка на малый стопор желтая",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 4.728,
    },
    {
      "avgPrice": 9.6,
      "id": 38226,
      "maxPrice": 9.6,
      "minPrice": 9.6,
      "name": "Дужка на малый стопор красная",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 19.2,
    },
    {
      "avgPrice": 6.12,
      "id": 38223,
      "maxPrice": 6.12,
      "minPrice": 6.12,
      "name": "Дужка на малый стопор черная",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 6.12,
    },
    {
      "avgPrice": 2.2,
      "id": 36883,
      "maxPrice": 2.2,
      "minPrice": 2.2,
      "name": "Карабин 7x70 А4",
      "ordersCount": 1,
      "totalCount": -3,
      "totalSum": 6.6,
    },
    {
      "avgPrice": 17.4193,
      "id": 35056,
      "maxPrice": 17.4193,
      "minPrice": 17.4193,
      "name": "29mm Carbo T2 блок",
      "ordersCount": 1,
      "totalCount": -3,
      "totalSum": 52.2579,
    },
    {
      "avgPrice": 0.4,
      "id": 34863,
      "maxPrice": 0.4,
      "minPrice": 0.4,
      "name": "Кольцо стопорное быстросъемное A4 1,5x20",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 0.4,
    },
    {
      "avgPrice": 2.1,
      "id": 34789,
      "maxPrice": 2.1,
      "minPrice": 2.1,
      "name": "Скоба плоская удлиненного типа A4 5mm",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 2.1,
    },
    {
      "avgPrice": 1.5,
      "id": 34786,
      "maxPrice": 1.5,
      "minPrice": 1.5,
      "name": "Скоба плоская укороченного типа 5mm A4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 1.5,
    },
    {
      "avgPrice": 6,
      "id": 34781,
      "maxPrice": 6,
      "minPrice": 6,
      "name": "Скоба с невыпадающим штифтом 10 mm A4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 6,
    },
    {
      "avgPrice": 4,
      "id": 34773,
      "maxPrice": 4,
      "minPrice": 4,
      "name": "Скоба под шестиграньник 10mm A4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 4,
    },
    {
      "avgPrice": 12,
      "id": 26503,
      "maxPrice": 12,
      "minPrice": 12,
      "name": "Обжим троса 6 мм",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 12,
    },
    {
      "avgPrice": 13,
      "id": 18171,
      "maxPrice": 13,
      "minPrice": 13,
      "name": "Нитки парусные парафинированный полиэстер",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 13,
    },
    {
      "avgPrice": 11.9032,
      "id": 12589,
      "maxPrice": 11.9032,
      "minPrice": 11.9032,
      "name": "16mm блок палубный",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 23.8064,
    },
    {
      "avgPrice": 10.4516,
      "id": 12581,
      "maxPrice": 10.4516,
      "minPrice": 10.4516,
      "name": "16mm Air блок с кольцом",
      "ordersCount": 1,
      "totalCount": -3,
      "totalSum": 31.3548,
    },
    {
      "avgPrice": 32.028,
      "id": 12527,
      "maxPrice": 32.028,
      "minPrice": 32.028,
      "name": "29mm Carbo блок двойной вертлюжный",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 32.028,
    },
    {
      "avgPrice": 29.6128,
      "id": 12410,
      "maxPrice": 29.6128,
      "minPrice": 29.6128,
      "name": "Стопор кулачковый средний алюминий",
      "ordersCount": 1,
      "totalCount": -4,
      "totalSum": 118.4512,
    },
    {
      "avgPrice": 0.2,
      "id": 8776,
      "maxPrice": 0.2,
      "minPrice": 0.2,
      "name": "Винт_полукруглая головка 6х40",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 0.4,
    },
    {
      "avgPrice": 6,
      "id": 8569,
      "maxPrice": 6,
      "minPrice": 6,
      "name": "Флаг Латвии 20х30см",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 6,
    },
    {
      "avgPrice": 18.67,
      "id": 8277,
      "maxPrice": 18.67,
      "minPrice": 18.67,
      "name": "Однокомпонентный лак VA3 0.35 л глянцевый",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 18.67,
    },
    {
      "avgPrice": 16,
      "id": 7526,
      "maxPrice": 16,
      "minPrice": 16,
      "name": "Шарнир удлинителя румпеля",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 32,
    },
    {
      "avgPrice": 84.4138,
      "id": 7300,
      "maxPrice": 84.4138,
      "minPrice": 84.4138,
      "name": "S19 I-Beam Car,Control Sheaves,6 Wheel",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 84.4138,
    },
    {
      "avgPrice": 4.4185,
      "id": 7191,
      "maxPrice": 4.4185,
      "minPrice": 4.4185,
      "name": "Блок одношкивный (Комплект из 2-х шт.) красный",
      "ordersCount": 1,
      "totalCount": -4,
      "totalSum": 17.674,
    },
    {
      "avgPrice": 99,
      "id": 6293,
      "maxPrice": 99,
      "minPrice": 99,
      "name": "Якорь адмиралтейский 20 кг (+веревка и цепь на складе)",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 99,
    },
    {
      "avgPrice": 1.7,
      "id": -201,
      "maxPrice": 1.7,
      "minPrice": 1.7,
      "name": "Палец стопорной 8x54 А4",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 1.7,
    },
  ],
  "rowsCount": 46,
}
`;
